clang/include/clang/Sema/Sema.h

```diff
@@ -13419,7 +13419,8 @@ FPGALoopAttrT *Sema::BuildSYCLIntelFPGALoopAttr(const AttributeCommonInfo &A,
                A.getParsedKind() ==
                    ParsedAttr::AT_SYCLIntelFPGAMaxInterleaving ||
                A.getParsedKind() ==
-                   ParsedAttr::AT_SYCLIntelFPGASpeculatedIterations) {
+                   ParsedAttr::AT_SYCLIntelFPGASpeculatedIterations ||
+               A.getParsedKind() == ParsedAttr::AT_SYCLIntelFPGALoopCountAvg) {
```

Comment:
Nothing for you to fix in your patch, but this is extending another anti-pattern where we have a templated function that still needs to look at the parse kind specifically (so the function looks generic but is not particularly generic).